src/librustc_mir/transform/uninhabited_enum_branching.rs

```diff
@@ -99,26 +99,21 @@ impl<'tcx> MirPass<'tcx> for UninhabitedEnumBranching {
             if let TerminatorKind::SwitchInt { values, targets, .. } =
                 &mut body.basic_blocks_mut()[bb].terminator_mut().kind
             {
-                let vals = &*values;
-                let zipped = vals.iter().zip(targets.iter());
-
-                let mut matched_values = Vec::with_capacity(allowed_variants.len());
-                let mut matched_targets = Vec::with_capacity(allowed_variants.len() + 1);
-
-                for (val, target) in zipped {
-                    if allowed_variants.contains(val) {
-                        matched_values.push(*val);
-                        matched_targets.push(*target);
-                    } else {
-                        trace!("eliminating {:?} -> {:?}", val, target);
-                    }
-                }
-
-                // handle the "otherwise" branch
-                matched_targets.push(targets.pop().unwrap());
-
-                *values = matched_values.into();
-                *targets = matched_targets;
+                // take otherwise out early
+                let otherwise = targets.pop().unwrap();
+                assert_eq!(targets.len(), values.len());
+                let mut i = 0;
+                targets.drain_filter(|_| {
+                    let should_rm = !allowed_variants.contains(&values[i]);
+                    i += 1;
+                    should_rm
+                });
+
+                let new_values: Vec<_> =
```

Comment:
why isn't this also using `drain_filter`?

Reply:
ah it's not a `Vec`, it's a `Cow<[u128]>`. I could take it to owned and `drain_filter` over it, would that be better?

Reply:
Ah, it's a `Cow`. You could match on it and take out the owned version if there is one and drain that, and otherwise do what you do here, that probably isn't worth the effor though, let's just keep what you wrote